pkg/operation/botanist/component/kubecontrollermanager/kube_controller_manager.go

```diff
@@ -555,29 +565,54 @@ func (k *kubeControllerManager) computeCommand(port int32) []string {
 		defaultHorizontalPodAutoscalerConfig = k.getHorizontalPodAutoscalerConfig()
 	)
 
+	podEvictionTimeout := metav1.Duration{Duration: 2 * time.Minute}
+	if v := k.config.PodEvictionTimeout; v != nil {
+		podEvictionTimeout = *v
+	}
+
+	nodeMonitorGracePeriod := metav1.Duration{Duration: 2 * time.Minute}
+	if v := k.config.NodeMonitorGracePeriod; v != nil {
+		nodeMonitorGracePeriod = *v
+	}
+
 	command = append(command,
 		"/usr/local/bin/kube-controller-manager",
-		"--allocate-node-cidrs=true",
 		"--attach-detach-reconcile-sync-period=1m0s",
-		"--controllers=*,bootstrapsigner,tokencleaner",
 		"--authentication-kubeconfig="+gardenerutils.PathGenericKubeconfig,
 		"--authorization-kubeconfig="+gardenerutils.PathGenericKubeconfig,
 		"--kubeconfig="+gardenerutils.PathGenericKubeconfig,
 	)
 
-	if k.config.NodeCIDRMaskSize != nil {
-		command = append(command, fmt.Sprintf("--node-cidr-mask-size=%d", *k.config.NodeCIDRMaskSize))
+	if !k.isWorkerless {
+		command = append(command,
+			"--allocate-node-cidrs=true",
+			"--controllers=*,bootstrapsigner,tokencleaner",
+			fmt.Sprintf("--cluster-cidr=%s", k.podNetwork.String()),
+			fmt.Sprintf("--cluster-signing-kubelet-client-cert-file=%s/%s", volumeMountPathCAClient, secrets.DataKeyCertificateCA),
+			fmt.Sprintf("--cluster-signing-kubelet-client-key-file=%s/%s", volumeMountPathCAClient, secrets.DataKeyPrivateKeyCA),
+			fmt.Sprintf("--cluster-signing-kubelet-serving-cert-file=%s/%s", volumeMountPathCAKubelet, secrets.DataKeyCertificateCA),
+			fmt.Sprintf("--cluster-signing-kubelet-serving-key-file=%s/%s", volumeMountPathCAKubelet, secrets.DataKeyPrivateKeyCA),
+			fmt.Sprintf("--horizontal-pod-autoscaler-downscale-stabilization=%s", defaultHorizontalPodAutoscalerConfig.DownscaleStabilization.Duration.String()),
+			fmt.Sprintf("--horizontal-pod-autoscaler-initial-readiness-delay=%s", defaultHorizontalPodAutoscalerConfig.InitialReadinessDelay.Duration.String()),
+			fmt.Sprintf("--horizontal-pod-autoscaler-cpu-initialization-period=%s", defaultHorizontalPodAutoscalerConfig.CPUInitializationPeriod.Duration.String()),
+			fmt.Sprintf("--horizontal-pod-autoscaler-sync-period=%s", defaultHorizontalPodAutoscalerConfig.SyncPeriod.Duration.String()),
+			fmt.Sprintf("--horizontal-pod-autoscaler-tolerance=%v", *defaultHorizontalPodAutoscalerConfig.Tolerance),
+			"--leader-elect=true",
+			fmt.Sprintf("--node-monitor-grace-period=%s", nodeMonitorGracePeriod.Duration),
+			fmt.Sprintf("--pod-eviction-timeout=%s", podEvictionTimeout.Duration),
+		)
+	}
+
+	if k.isWorkerless {
+		command = append(command,
+			"--controllers=namespace,serviceaccount,serviceaccount-token,clusterrole-aggregation,garbagecollector,csrapproving,csrcleaner,csrsigning,bootstrapsigner,tokencleaner,resourcequota",
```

Comment:
care should be taken that any new controller introduced should be added to the list here. Easy to miss. You should update the documentation if there is any or define a constant which has the allowed set of controllers in case of workerless.